crates/pathfinder/src/state/sync.rs

```diff
@@ -666,7 +666,10 @@ async fn l2_reorg(
         // TODO: clean up state tree's as well...
 
         StarknetBlocksTable::reorg(&transaction, reorg_tail)
-            .context("Delete L2 state from database")?;
+            .context("Delete L2 blocks from database")?;
+
+        StarknetStateUpdatesTable::reorg(&transaction, reorg_tail)
+            .context("Delete L2 state updates from database")?;
```

Comment:
Why do we need this?

We're using `ON CASCADE DELETE` in the database schema so deleting the blocks from the DB will automatically delete the state update from the DB?

Reply:
Actually we're not, I did a fixup for that in https://github.com/eqlabs/pathfinder/pull/484/commits/7ad39c42d128f43d9e4f1f85f2d20885dcead208 but did not separate it into a proper fixup! commit, so it did not get squashed correctly.

Reply:
Hmm, but why not use `ON CASCADE DELETE` instead of this?

Reply:
Following a discussion with @kkovaacs
- at the moment we cannot use foreign key because `hash` is `starknet_blocks` is neither PK nor `UNIQUE`
- adding `UNIQUE` does not make sense right now because the schema will be significantly restructured after forks are allowed
- `StarknetStateUpdatesTable::reorg` is sufficient right now.